weblate/trans/models/component.py

```diff
@@ -871,6 +879,9 @@ def save(self, *args, **kwargs) -> None:
         self.intermediate = cleanup_path(self.intermediate)
         self.new_base = cleanup_path(self.new_base)
 
+        if self.key_filter_re != self.key_filter:
```

Comment:
This seems wrong, you are comparing a compiled regular expression with a string what doesn't work:

```pyconsole
>>> import re
>>> re.compile('.*') == '.*'
False
```

Compare in `if self.id` branch above with a `old` instance.